app/controllers/administrative-units/administrative-unit/core-data/edit.js

```diff
@@ -55,6 +47,32 @@ export default class AdministrativeUnitsAdministrativeUnitCoreDataEditController
       }
     }
 
+    let identifiers = yield this.administrativeUnit.identifiers;
+    if (
+      identifiers.firstObject.structuredIdentifier.get('hasDirtyAttributes')
+    ) {
+      let structuredIdentifier = yield identifiers.firstObject
+        .structuredIdentifier;
+
+      let isNewIdentifier = structuredIdentifier.get('isNew');
+      yield structuredIdentifier.save();
+
+      if (isNewIdentifier) {
+        yield identifiers.firstObject.save();
+      }
+    }
+    if (identifiers.lastObject.structuredIdentifier.get('hasDirtyAttributes')) {
```

Comment:
This seems to be the same code as for the other one? I think you can loop over all the items instead?